### ISSUE TITLE
Drachtio server crash

### ISSUE THREAD

**Issue by xquanluu** [drachtio/drachtio-server#390]
```
[Current thread is 1 (Thread 0x7fadab01c540 (LWP 3218875))]
(gdb) bt
#0  __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at ./nptl/pthread_kill.c:44
#1  0x00007fadab0a9e8f in __pthread_kill_internal (signo=6, threadid=<optimized out>) at ./nptl/pthread_kill.c:78
#2  0x00007fadab05afb2 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#3  0x00007fadab045472 in __GI_abort () at ./stdlib/abort.c:79
#4  0x00007fadaae9d919 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00007fadaaea8e1a in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00007fadaaea8e85 in std::terminate() () from /lib/x86_64-linux-gnu/libstdc++.so.6
#7  0x000055ccf697b159 in std::__terminate () at /usr/include/x86_64-linux-gnu/c++/12/bits/c++config.h:312
#8  std::thread::~thread (this=0x55ccf9147968, __in_chrg=<optimized out>) at /usr/include/c++/12/bits/std_thread.h:151
#9  drachtio::RequestHandler::~RequestHandler (this=0x55ccf9147950, __in_chrg=<optimized out>) at ../src/request-handler.cpp:442
#10 0x000055ccf697f652 in std::_Sp_counted_ptr<drachtio::RequestHandler*, (__gnu_cxx::_Lock_policy)2>::_M_dispose (this=<optimized out>) at /usr/include/c++/12/bits/shared_ptr_base.h:428
#11 0x000055ccf689128a in std::_Sp_counted_base<(__gnu_cxx::_Lock_policy)2>::_M_release_last_use (this=0x55ccf923dae0) at /usr/include/c++/12/bits/shared_ptr_base.h:175
#12 std::_Sp_counted_base<(__gnu_cxx::_Lock_policy)2>::_M_release_last_use_cold (this=0x55ccf923dae0) at /usr/include/c++/12/bits/shared_ptr_base.h:199
#13 0x000055ccf697d609 in std::__shared_count<(__gnu_cxx::_Lock_policy)2>::~__shared_count (this=0x7fff426c5e38, __in_chrg=<optimized out>) at /usr/include/c++/12/bits/shared_ptr_base.h:1071
#14 std::__shared_ptr<drachtio::RequestHandler, (__gnu_cxx::_Lock_policy)2>::~__shared_ptr (this=0x7fff426c5e30, __in_chrg=<optimized out>) at /usr/include/c++/12/bits/shared_ptr_base.h:1524
#15 std::shared_ptr<drachtio::RequestHandler>::~shared_ptr (this=0x7fff426c5e30, __in_chrg=<optimized out>) at /usr/include/c++/12/bits/shared_ptr.h:175
#16 drachtio::close_socket (clientp=<optimized out>, item=<optimized out>) at ../src/request-handler.cpp:417
#17 0x00007fadabc3728d in ?? () from /lib/x86_64-linux-gnu/libcurl.so.4
#18 0x00007fadabc37355 in ?? () from /lib/x86_64-linux-gnu/libcurl.so.4
#19 0x00007fadabc3ad45 in ?? () from /lib/x86_64-linux-gnu/libcurl.so.4
#20 0x00007fadabc3ab89 in ?? () from /lib/x86_64-linux-gnu/libcurl.so.4
#21 0x00007fadabc8df58 in ?? () from /lib/x86_64-linux-gnu/libcurl.so.4
#22 0x00007fadabc3a6d9 in ?? () from /lib/x86_64-linux-gnu/libcurl.so.4
#23 0x00007fadabc6e120 in curl_multi_cleanup () from /lib/x86_64-linux-gnu/libcurl.so.4
#24 0x000055ccf697afc7 in drachtio::RequestHandler::~RequestHandler (this=0x55ccf9147950, __in_chrg=<optimized out>) at ../src/request-handler.cpp:441
#25 0x000055ccf697f652 in std::_Sp_counted_ptr<drachtio::RequestHandler*, (__gnu_cxx::_Lock_policy)2>::_M_dispose (this=<optimized out>) at /usr/include/c++/12/bits/shared_ptr_base.h:428
#26 0x000055ccf689128a in std::_Sp_counted_base<(__gnu_cxx::_Lock_policy)2>::_M_release_last_use (this=0x55ccf923dae0) at /usr/include/c++/12/bits/shared_ptr_base.h:175
#27 std::_Sp_counted_base<(__gnu_cxx::_Lock_policy)2>::_M_release_last_use_cold (this=0x55ccf923dae0) at /usr/include/c++/12/bits/shared_ptr_base.h:199
#28 0x00007fadab05d55d in __run_exit_handlers (status=0, listp=0x7fadab1f1820 <__exit_funcs>, run_list_atexit=run_list_atexit@entry=true, run_dtors=run_dtors@entry=true) at ./stdlib/exit.c:116
#29 0x00007fadab05d69a in __GI_exit (status=<optimized out>) at ./stdlib/exit.c:146
#30 0x000055ccf68976e5 in drachtio::DrachtioController::handleSigTerm (this=0x55ccf9192000, signal=<optimized out>) at ../src/controller.cpp:357
#31 <signal handler called>
#32 0x00007fadab127e26 in epoll_wait (epfd=8, events=events@entry=0x7fff426c7370, maxevents=4, timeout=13328) at ../sysdeps/unix/sysv/linux/epoll_wait.c:30
#33 0x000055ccf6a101ac in su_epoll_port_wait_events (self=0x55ccf91a9180, tout=<optimized out>) at su_epoll_port.c:495
#34 0x000055ccf6a0fc7a in su_base_port_run (self=0x55ccf91a9180) at su_base_port.c:349
#35 0x000055ccf68a3366 in drachtio::DrachtioController::run (this=0x55ccf9192000) at ../src/controller.cpp:1391
#36 0x000055ccf6885bc9 in main (argc=28, argv=0x7fff426c7878) at ../src/main.cpp:47
```

**Comment by xquanluu** [comment]
```
2024-11-20 08:24:55.196019 Client::read_handler read: 58ad4fa2-0693-4518-b4b9-e0b068c69f8f|sip|ad880f42-cb88-41b0-ace8-1afb2325a4ba|^M
SIP/2.0 200 OK^M
Call-ID: c3599790-21bb-123e-dcba-0630fe05ec17^M
cseq: 91491755 OPTIONS^M
from: <sip:10.9.32.156:5070>;tag=2p2jatv14Zrpj^M
to: <sip:10.9.32.156>^M
Content-Length: 0^M
X-Members: 1^M
^M

2024-11-20 08:24:55.196040 Client::processMessage - got request with 4 tokens
2024-11-20 08:24:55.196044 Client::processMessage - request id 58ad4fa2-0693-4518-b4b9-e0b068c69f8f, request type: sip transaction id: ad880f42-cb88-41b0-ace8-1afb2325a4ba, dialog id:
2024-11-20 08:24:55.196048 ClientController::addApiRequest: clientMsgId 58ad4fa2-0693-4518-b4b9-e0b068c69f8f; size: 1
2024-11-20 08:24:55.196083 SipDialogController::doRespondToSipRequest thread 140698548172096
2024-11-20 08:24:55.196093 SipDialogController::findAndRemoveTransactionIdForIncomingRequest - searching transactionId ad880f42-cb88-41b0-ace8-1afb2325a4ba
2024-11-20 08:24:55.196098 SipDialogController::findAndRemoveTransactionIdForIncomingRequest - failed to find transactionId ad880f42-cb88-41b0-ace8-1afb2325a4ba, most likely this is a response to an invite we sent
2024-11-20 08:24:55.196118 pending-request: removing entry, prior to removal length: 1
2024-11-20 08:24:55.196122 pending-request: removed entry, timer not set (queue is empty after removal), length: 0
2024-11-20 08:24:55.196134 SipDialogController::addIncomingRequestTransaction - adding transactionId ad880f42-cb88-41b0-ace8-1afb2325a4ba for irq:0x555c3935ec80
2024-11-20 08:24:55.196140 SipDialogController::findAndRemoveTransactionIdForIncomingRequest - searching transactionId ad880f42-cb88-41b0-ace8-1afb2325a4ba
2024-11-20 08:24:55.196144 SipDialogController::doRespondToSipRequest found incoming transaction 0x555c3935ec80
2024-11-20 08:24:55.196187 SipTransport::getContactUri - created Contact header: sip:40.172.25.99:5060
2024-11-20 08:24:55.196259 makeTags - Adding well-known header 'Call-ID' with value 'c3599790-21bb-123e-dcba-0630fe05ec17'
2024-11-20 08:24:55.196266 makeTags - Adding well-known header 'cseq' with value '91491755 OPTIONS'
2024-11-20 08:24:55.196272 makeTags - Adding well-known header 'from' with value '<sip:10.9.32.156:5070>;tag=2p2jatv14Zrpj'
2024-11-20 08:24:55.196279 makeTags - Adding well-known header 'to' with value '<sip:10.9.32.156>'
2024-11-20 08:24:55.196287 makeTags - custom header: 'X-Members', value: 1
2024-11-20 08:24:55.196319 nta.c:6970 incoming_reply() incoming_reply: tport_tsend: error (No buffer space available) while sending 200 OK for OPTIONS (91491755)
2024-11-20 08:24:55.196325 SipDialogController::doRespondToSipRequest destroying irq 0x555c3935ec80
2024-11-20 08:24:55.196328 ClientController::removeApiRequest: clientMsgId 58ad4fa2-0693-4518-b4b9-e0b068c69f8f; size: 0
2024-11-20 08:24:55.196337 ClientController::removeNetTransaction: transactionId ad880f42-cb88-41b0-ace8-1afb2325a4ba; size: 0
2024-11-20 08:24:55.196350 nta.c:6970 incoming_reply() incoming_reply: tport_tsend: error (No buffer space available) while sending 500 Internal Server Error for OPTIONS (91491755)
```

**Comment by xquanluu** [comment]
close this issue due to we have not put udp interface as a first options in drachtio.service file
